public-ui/src/app/data-browser/components/source-tree/source-tree-react.component.tsx

```diff
@@ -0,0 +1,194 @@
+        width: '.5rem',
+        height: '.5rem'
+    },
+    loading: {
+        marginTop: '0.1rem',
+        left: '-.1rem'
+    }
+});
+
+
+interface ChildState {
+    isChildSelected: boolean;
+    children: any;
+}
+interface ChildProps {
+    id: number;
+}
+
+const ChildFetchingComponent = (
+    class extends React.Component<ChildProps, ChildState> {
+        constructor(props: ChildProps) {
+            super(props);
+            this.state = {
+                isChildSelected: false,
+                children: undefined
+            };
+        }
+
+        childClick() {
+            this.getChildren();
+            this.setState({
+                isChildSelected: !this.state.isChildSelected
+            });
+        }
+
+        getChildren() {
+            return dataBrowserApi().getCriteriaChildren(this.props.id).then(
+                (data) => {
+                    this.setState({ children: data.items });
+
+                }
+            );
+        }
+
+        componentDidMount() {
+            this.getChildren();
+        }
+
+        render() {
+            const { children } = this.state;
+            return <React.Fragment>
+                {children && children.map((child, index) => {
+                    return <div key={index} style={styles.childNode}>
+                        {/* tslint:disable-next-line:no-use-before-declare */}
+                        <SourceTreeComponent node={child} />
+                    </div>;
+                })}
+            </React.Fragment>;
+        }
+
+    });
+
+interface SourceTreeProps {
+    node: any;
+    first?: boolean;
+}
+
+interface SourceTreeState {
+    isConceptSelected: boolean;
+    isHandelSelected: boolean;
+    highlightId: number;
+}
+
+export const SourceTreeComponent = (
+    class extends React.Component<SourceTreeProps, SourceTreeState> {
+        constructor(props: SourceTreeProps) {
+            super(props);
+            this.state = {
+                isConceptSelected: false,
+                isHandelSelected: undefined,
+                highlightId: undefined
+            };
+        }
+        conceptClick() {
+            localStorage.setItem('treeHighlight', this.props.node.id);
+            this.setState({
+                isConceptSelected: !this.state.isConceptSelected
+            });
+            this.getTreeHighlight();
+        }
+        getTreeHighlight() {
+            this.setState({
+                highlightId: parseInt(localStorage.getItem('treeHighlight') , 10)
+            });
+        }
+        handleClick() {
+            this.setState({
+                isHandelSelected: !this.state.isHandelSelected
+            });
+
+        }
+        componentDidMount() {
+            // this.getTreeHighlight();
+            this.setState({ isHandelSelected: this.props.first });
+
+        }
+
+        render() {
+            const { node } = this.props;
+            const { isHandelSelected, isConceptSelected, highlightId } = this.state;
+
```

Comment:
```suggestion
            const nodeChildren = node.children || children;
```
I couldn't find any case where `node.children` is set. Is there a scenario where `children` is already set on the `node` prop?

Reply:
If node as children it's an array with the child sources. If it doesn't they can be fetched.